Emby.Server.Implementations/Sorting/AiredEpisodeOrderComparer.cs

```diff
@@ -156,8 +146,14 @@ private static int CompareEpisodes(Episode x, Episode y)
         {
             var xValue = ((x.ParentIndexNumber ?? -1) * 1000) + (x.IndexNumber ?? -1);
             var yValue = ((y.ParentIndexNumber ?? -1) * 1000) + (y.IndexNumber ?? -1);
+            var comparisonResult = xValue.CompareTo(yValue);
+            // If equal, compare premiere dates
+            if (comparisonResult == 0 && x.PremiereDate.HasValue && y.PremiereDate.HasValue)
+            {
+                comparisonResult = DateTime.Compare(x.PremiereDate.Value, y.PremiereDate.Value);
```

Comment:
What if there's slight timezone differences? Should the timestamp matter?

Reply:
Are you asking about x.Timestamp or the PremiereDate as a timestamp since it is Datetime? If the former I am not sure how Timestamp is used but I don't think it has to do with the time the episode was premiered and is more used for encoding. If the later, I am unsure how the sorter would do anything more. The timezone would need to be supplied by the metadata server and be a part of PremiereDate when written to the database.

Reply:
I'm asking about the case where the premiere date is coming from different metadata providers where the timestamp may differ.

What I'm asking is if the timestamp is even relevant?

Reply:
I don't believe this would affect things. The timezone if one is available will be accounted for in the comparison. If the agent isn't recording the timezone information or the database provides incorrect data then the sorting function is not at fault here and cannot be made to fix that problem.

Reply:
I guess that's fair.